lib/core/bucketEntryShards/MongoDBBucketEntryShardsRepository.ts

```diff
@@ -1,47 +1,69 @@
-import { BucketEntry } from "../bucketEntries/BucketEntry";
-import { BucketEntryShard } from "./BucketEntryShard";
-import { BucketEntryShardsRepository } from "./Repository";
+import { BucketEntry } from '../bucketEntries/BucketEntry';
+import { BucketEntryShard } from './BucketEntryShard';
+import { BucketEntryShardsRepository } from './Repository';
 
-export class MongoDBBucketEntryShardsRepository implements BucketEntryShardsRepository {
+const formatFromMongoToBucketEntryShard = (
+  mongoBucketEntryShard: any
+): BucketEntryShard => {
+  const id = mongoBucketEntryShard._id.toString();
+  const bucketEntryShard = mongoBucketEntryShard.toObject();
+  delete bucketEntryShard._id;
+  return {
+    ...bucketEntryShard,
+    id: id,
+    bucketEntry: mongoBucketEntryShard.bucketEntry.toString(),
+    shard: mongoBucketEntryShard.shard.toString(),
+  };
+};
+export class MongoDBBucketEntryShardsRepository
+  implements BucketEntryShardsRepository
+{
   constructor(private model: any) {}
-  
-  find(where: Partial<BucketEntryShard>): Promise<BucketEntryShard[]> {
-    return this.model.find(where);
-  }
 
-  findByBucketEntry(bucketEntryId: BucketEntry['id']): Promise<BucketEntryShard[] & { sort: (...x: any) => any; }> {
-    return this.model.find({ bucketEntry: bucketEntryId });
+  async find(where: Partial<BucketEntryShard>): Promise<BucketEntryShard[]> {
```

Comment:
Be careful here when `id` is used here, as Mongo seems to need `_id` instead or exclude id from the `Partial` where to force the findById usage